src/main/kotlin/org/arctgkolbasy/bot/handler/BillCommand.kt

```diff
@@ -0,0 +1,116 @@
+package org.arctgkolbasy.bot.handler
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.github.kotlintelegrambot.Bot
+import com.github.kotlintelegrambot.entities.TelegramFile
+import com.github.kotlintelegrambot.entities.Update
+import emoji.Emoji
+import org.arctgkolbasy.bot.user.User
+import org.arctgkolbasy.bot.user.UserRoles
+import org.arctgkolbasy.product.ProductRepository
+import org.arctgkolbasy.user.UserService
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.stereotype.Controller
+
+@Controller
+class BillCommand(
+    userService: UserService,
+    @Qualifier("currentUserHolder")
+    currentUserHolder: ThreadLocal<User?>,
+    val productRepository: ProductRepository,
+    val objectMapper: ObjectMapper,
+) : SecuredCommand(
+    userService = userService,
+    currentUserHolder = currentUserHolder,
+    isStateless = false,
+) {
+    override fun getCommandName(): String = BILL
+
+    override fun sessionCheck(sessionKey: String?, session: String?): Boolean = sessionKey in steps
+
+    override fun checkUserAccess(user: User): Boolean = UserRoles.USER in user.roles
+
+    override fun handleUpdateInternal(user: User, bot: Bot, update: Update) = when (user.sessionKey) {
+        InputId.STEP_1_INPUT_ID.step -> findBillById(update, bot, user)
+        else -> stepOneInputIdOrSendPhoto(update, bot, user)
+    }
+
+    private fun findBillById(update: Update, bot: Bot, user: User) {
+        if (update.message().startsWith("/")) {
+            user.clearSession()
+            stepOneInputIdOrSendPhoto(update, bot, user)
+        } else {
+            val idBill = update.message()
+            try {
+                idBill.toLong()
+            } catch (e: Exception) {
+                user.clearSession()
+                bot.sendMessage(update.chatIdUnsafe(), "Неправильный $ID_EMOJI. Введи корректный:")
+                user.updateSession(
+                    InputId.STEP_1_INPUT_ID.step,
+                    objectMapper.writeValueAsString(InputIdSession())
+                )
+                return
+            }
+            if (productRepository.findAll().any { it.id == idBill.toLong() }) {
+                sendPhoto(bot, update, idBill, user)
+            } else {
+                bot.sendMessage(
+                    update.chatIdUnsafe(),
+                    "Нет такого продукта$NON_EMOJI.\nВведи еще раз или посмотри список продуктов$LOUPE_EMOJI /product_in_stock"
+                )
+                user.updateSession(
+                    InputId.STEP_1_INPUT_ID.step,
+                    objectMapper.writeValueAsString(InputIdSession())
+                )
```

Comment:
опять таки чтобы заретраить ошибку достаточно выбросить эксепшн. еще не стоит хардкодить команду, тк если ты ее потом поменяешь, надо будет незабыть изменить ее здесь
```suggestion
                throw IllegalArgumentException("Нет такого продукта${Emoji.PERSON_SHRUGGING.emoji}.\nВведи еще раз или посмотри список продуктов${Emoji.MAGNIFYING_GLASS_TILTED_LEFT.emoji} /product_in_stock")
```